elastic/beats
# Osquerybeat: Support custom namespaces for scheduled queries

## What does this PR do?

This PR adds a proper handling for the custom namespaces for  osquery.
By default the namespace for osquery_manager integration is set to "default" and all the osquery results are stored in ```logs-osquery_manager.result-default``` data stream.
This allows the user to change it, for example to ```custom``` which will have osquery results to be sent to ```logs-osquery_manager.result-custom``` datastream.

This addresses the osquerybeat side for https://github.com/elastic/security-team/issues/1598

## Why is it important?

Handles the non-default namespace per osquery_manager integration correctly.

## Checklist


## How to test this PR locally

1. Add two or more osquery_manager integrations to the policy with queries and possibly change the namespaces on some of them to something else than ```default```
2. Observe the osquery results data is sent to the configured namespace

## Related issues

- Relates https://github.com/elastic/security-team/issues/1598

## Logs

Osquerybeat log showing documents sent to the different datastream (different namespaces)
```
{"log.level":"info","@timestamp":"2021-08-18T16:13:35.124-0400","log.logger":"osquerybeat","log.origin":{"file.name":"beater/osquerybeat.go","file.line":486},"message":"1 events sent to index logs-osquery_manager.result-custom","service.name":"osquerybeat","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-08-18T16:13:35.239-0400","log.logger":"osquerybeat","log.origin":{"file.name":"beater/osquerybeat.go","file.line":486},"message":"3 events sent to index logs-osquery_manager.result-default","service.name":"osquerybeat","ecs.version":"1.6.0"}  
```

## Screenshots
A part of the policy that shows permissions for a couple two different integration with ```default``` and ```custom``` namespaces.
<img width="468" alt="Screen Shot 2021-08-18 at 3 44 02 PM" src="https://user-images.githubusercontent.com/872351/129964721-2e7b98c6-9c96-40d8-9e7e-4bab8bc21258.png">

The ```default``` namespace was collecting ```select * from users``` results.
<img width="1376" alt="Screen Shot 2021-08-18 at 3 44 24 PM" src="https://user-images.githubusercontent.com/872351/129964900-64fa4e64-6222-47e0-91f3-ee9771398cfb.png">

The ```custom``` namespace was collecting ```select * from upstream``` results.
<img width="1363" alt="Screen Shot 2021-08-18 at 3 44 44 PM" src="https://user-images.githubusercontent.com/872351/129965011-c6909846-adef-428b-b526-fc9a6c3e1c64.png">
